### ISSUE TITLE
Caching hits the network even when it's not needed

### ISSUE THREAD

**Issue by indirect** [bundler/bundler#3670]
With a previously-installed bundle, there shouldn't be any need to hit the network to cache the gems that are already installed. In spite of that, install completes and then cache hits the network:

```
$ bundle pack
Using bundler 1.9.9
Using rake 10.4.2
Using rack 1.6.1
Bundle complete! 2 Gemfile dependencies, 3 gems now installed.
Bundled gems are installed into ./.bundle.
Updating files in vendor/cache
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
  * rake-10.4.2.gem
  * rack-1.6.1.gem
```

**Comment by segiddins** [comment]
I cannot reproduce this, unless I'm missing something?

**Comment by indirect** [comment]
Maybe it's fixed on master? I just saw it again today while running `bundle pack` on 1.9.9.

**Comment by segiddins** [comment]
@indirect closing since you never provided steps to reproduce ;)

**Comment by indirect** [comment]
Steps to reproduce on 1.10.6:

```
$ bundle init
$ echo "gem 'rack'" >> Gemfile
$ echo "gem 'rake'" >> Gemfile
$ bundle install --path .bundle
# note that this installation contacts rubygems.org
$ bundle package
# note that this _also_ contacts rubygems.org, even though the gems are already local and the lockfile is already present
```

**Comment by indirect** [comment]
Here's the output that doesn't make sense while running pack:

```
$ bundle pack
Using bundler 1.10.6
Using rack 1.6.4
Using rake 10.4.2
Bundle complete! 2 Gemfile dependencies, 3 gems now installed.
Bundled gems are installed into ./.bundle.
Updating files in vendor/cache
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
  * rake-10.4.2.gem
  * rack-1.6.4.gem
```

The specific general idea here is that if `install` can run without contacting rubygems.org, than so can `pack`. There's no reason to go get gem metadata again.

**Comment by segiddins** [comment]
ok, thanks for the repro steps!

**Comment by agis-** [comment]
I've just followed the repro steps and couldn't reproduce the issue.

**Comment by indirect** [comment]
Great, this is probably fixed by 1.11 then.

**Comment by TimMoore** [comment]
Sounds like we can close this
